webhooks/networking/ingress_validator.go

```diff
@@ -88,6 +98,21 @@ func (v *ingressValidator) ValidateDelete(ctx context.Context, obj runtime.Objec
 	return nil
 }
 
+// checkIngressClass checks to see if this ingress is handled by this controller.
+func (v *ingressValidator) checkIngressClass(ctx context.Context, ing *networking.Ingress) (bool, error) {
```

Comment:
nit: the `checkIngressClass` name is confusing. Since the purpose here is to check if the webhook should skip validating further, we could rename to either of `checkIngressSkipValidation`, `shouldSkipFurtherValidation`, `checkIngressUnmanaged`.

Changes look good otherwise.

Reply:
This function is validating the `kubernetes.io/ingress.class` annotation, the `IngressClassName` field, and the `spec.controller` of the `IngressClass` referenced by the `IngressClassName` field. Skipping further validation is only one possible outcome; it can fail validation if the `IngressClassName` references a nonexistent `IngressClass`.

For naming these sub-validation methods, I tend to prefer using the name of the field or subsection that is being validated.  The existing naming pattern isn't entirely consistent; this could be fixed in follow-up.

It looks like `checkIngressClassUsage()` is being made redundant; it could be removed in this or the followup PR.

Reply:
lets refactor in a follow up PR